### DISCUSSION TITLE
How do I make array of enum values out of the enum generated by Prisma?

### DISCUSSION THREAD

**Discussion by jezsung** [prisma/prisma#10617] [Q&A]
I'm looking for the simplest solution. The one possible solution I came up with is
```
// Let's say the name of the enum is Roles
Object.keys(Roles);
```
But the generated array is an array of string. It doesn't fully utilize Typescript. I want it to be an array of the enum type.

**Comment by austincrim** [comment]
Hey @jezsung 👋,

When you say:
> I want it to be an array of the enum type

what exactly do you want to be in the array? Do you want an array of the possible enum values?

**Reply by jezsung** [reply]
Yes, and I want those values to be the type of enum.

**Reply by austincrim** [reply]
Does `Object.values(Roles)` give you the desired type?

**Comment by xolesteroll** [comment]
> Does `Object.values(Roles)` give you the desired type?
Hello, i ve tried that. but in my case i get and error TypeError: Cannot set properties of undefined (setting 'Role').
Why can that be happenning? 

![image](https://github.com/prisma/prisma/assets/67501086/ab34a595-302b-483f-b376-9f5eddce2cdb)
![image](https://github.com/prisma/prisma/assets/67501086/86c99593-faa4-4bb1-82d1-6b2e1fc2fdb7)
![image](https://github.com/prisma/prisma/assets/67501086/39a4284a-44b1-4327-83e8-76031b9f6a34)

**Comment by mikelneonedwin** [comment]
@jezsung this is a very crude example but it helped me

```js
// path: prisma/generators/enum-array.ts
import pkg from "@prisma/generator-helper";
import { writeFileSync } from "node:fs";
import { resolve } from "node:path";

const { generatorHandler } = pkg;

generatorHandler({
  onManifest() {
    return {
      defaultOutput: "../generated/enums.ts",
      prettyName: "Prisma Enums Generator",
    };
  },
  async onGenerate(options) {
    // Build JS content
    let content = "// THIS FILE IS AUTO-GENERATED FROM PRISMA ENUMS\n\n";

    options.dmmf.datamodel.enums.forEach((enm) => {
      const name = enm.name;
      const values = enm.values.map((v) => `"${v.name}"`).join(", ");
      content += `export const ${name}EnumArray = [${values}] as const;\n\n`;
    });

    writeFileSync(resolve("generated", "enums.ts"), content, "utf-8");
  },
});
```

If you have this in your schema

```prisma
// path: prisma/schema.prisma
generator enums {
  provider = "tsx ./prisma/generators/enum-array.ts"
}

enum BlogStatus {
  published
  drafted
}

enum BlogCategory {
  Company
  Product
  Tutorial
  Engineering
  Opinion
  Research
  Launch
  Culture
  Marketing
  Event
  FAQ
  Storytelling
  Policy
  Playground
}
```

Once you generate, you get the resulting code...

```ts
// path: generated/enums.ts
// THIS FILE IS AUTO-GENERATED FROM PRISMA ENUMS

export const BlogStatusEnumArray = ["published", "drafted"] as const;

export const BlogCategoryEnumArray = ["Company", "Product", "Tutorial", "Engineering", "Opinion", "Research", "Launch", "Culture", "Marketing", "Event", "FAQ", "Storytelling", "Policy", "Playground"] as const;
```

I hope this helps out
